jsonrpc/eth_endpoint.go

```diff
@@ -375,6 +431,62 @@ func (e *Eth) GetTransactionReceipt(hash types.Hash) (interface{}, error) {
 	return toReceipt(raw, txn, uint64(txIndex), block.Header, logs), nil
 }
 
+// GetBlockReceipts returns all transaction receipts for a given block.
+func (e *Eth) GetBlockReceipts(number BlockNumber) (interface{}, error) {
+	num, err := GetNumericBlockNumber(number, e.store)
+	if err != nil {
+		return nil, err
+	}
+
+	block, ok := e.store.GetBlockByNumber(num, true)
+	if !ok {
+		return nil, nil
+	}
+
+	blockHash := block.Hash()
+	if len(block.Transactions) == 0 {
+		e.logger.Warn(
+			fmt.Sprintf("No transactions found for block with hash [%s]", blockHash.String()),
+		)
+
+		return nil, nil
+	}
+
+	receipts, errR := e.store.GetReceiptsByHash(blockHash)
+	if errR != nil {
+		// block receipts not found
+		e.logger.Warn(
+			fmt.Sprintf("Receipts for block with hash [%s] not found", blockHash.String()),
+		)
+
+		return nil, errR
+	}
+
+	numberOfReceipts := len(receipts)
+	if numberOfReceipts == 0 {
+		// Receipts not written yet on the db
+		e.logger.Warn(
+			fmt.Sprintf("No receipts found for block with hash [%s]", blockHash.String()),
+		)
+
+		return nil, nil
+	}
+
+	resReceipts := make([]*receipt, numberOfReceipts)
+	logIndex := 0
+
+	for i, transaction := range block.Transactions {
+		raw := receipts[i]
+		// accumulate receipt logs indexes from block transactions
+		// that are before the desired transaction
+		logIndex += len(receipts[i].Logs)
```

Comment:
I think this `logIndex` should be incremented after the `toReceipt` line, as it is now, the first receipt logs will start from a non zero index.